### ISSUE TITLE
Scheme reports/forms section: "Scheme Status"

### ISSUE THREAD

**Issue by robgaston** [archesproject/arches-lingo#116]
# preamble

to build out the sections of the scheme forms/reports we will need to:

1. define mappings between form/report fields and the Lingo models
2. implement backend serializers/APIs
3. implement front end UI

# mapping issues

## scheme status

![scheme status](https://github.com/user-attachments/assets/092fbba3-4f8e-4a57-894c-04b0cac0b6cd)

The "Scheme Status" section appears to contain data from multiple nodegroups as well as resource lifecycle information.

The actual "status" values "Scheme Edit Status" and "Scheme Publish Status" probably map onto lifecycle states. questions:

- is it correct that there is no semantic representation of this value(s)?
- should these actually be one thing, ie lifecycle status?
- does it actually make sense to edit these values here? thinking maybe they are read only here in the report/form and we might have a dedicated UI for changing lifecycle status

## scheme names

![scheme name:label](https://github.com/user-attachments/assets/ba10474a-6243-4848-b021-b13c71d6fafb)

"Scheme Name" seems to come from the label nodegroup, but this does not seem to account for alt labels or localization, so what should we do here?

 ## scheme languages

![scheme languages:language](https://github.com/user-attachments/assets/f354356f-7414-4e8e-ab36-3362534e621a)scheme

"Scheme Languages" appears to be stored in its own cardinality 1 nodegroup, which makes sense, but means the serializer will need to pull from multiple tiles.

## scheme creator

![scheme creator:creation metadata](https://github.com/user-attachments/assets/eb99865d-d5f6-418b-97b8-ddfceb9765fd)

"Scheme Owner" is stored alongside other creation metadata in a tile w/ it's own nodegroup, again the serializer will need to conflate these data accordingly.

@Habennin @Philbox @dwuthrich we should discuss & resolve the above questions as they will be blockers to development here. thanks!

**Comment by Philbox** [comment]
@robgaston not sure about this one at all.
In my experience schemes don't usually have altlabels but I could certainly see a use for them - AAT vs Art and Architecture Thesaurus etc.

I think the status is probably a lifecycle issue.

@habennin may have thoughts

**Comment by Habennin** [comment]
scheme names, yes we modelled it with the same data structure as concepts to be consistent in the data and so that there could also be a consistent user interface feel.

re scheme language: yes this is the place where that data is foreseen to land "text_language" field. Not sure I understand the serializer part.

re scheme owner I'm not sure if that is the creator or the rights holder. I would think that it might be the rights holder in which case it should could in the rights node group as rights_holder or maybe it is also in creation / creation_actor. Not sure I understand I understand what it means that the serializer has to conflate the data.

re status: I think I meant to add a node for that but never got around to it. I am pretty sure we have a status node on the concept. I mean the function would be that the editor could flip a switch and say what status they think the thing is in. Should they be able to do this? Dunno. Or maybe the system auto populates it. Still seems like a good idea to have it as semantic data. I can add that node if the concensus is that it is a good idea.

**Comment by robgaston** [comment]
should we then just simplify the model so that schemes just have a single name?

we need to decide definitively what “scheme owner” is and where to store is semantically.  thoughts, @dwuthrich?

as far as status goes, we should add that semantic node if we want to store it.  This would still not necessarily imply that status should be editable here and it doesn’t seem to make sense for it to be two nodes.  I think we should treat lifecycle management outside this form.  thoughts, @dwuthrich?

**Comment by robgaston** [comment]
Based on conversations with the team:

- Status fields can be removed from this form; status should be shown in the report, but will be implemented as part of lifecycle status, ie future work outside this form interaction
- the scheme owner field can be removed; this should be auto-assigned on the backend via Django.  currently this is a resource-instance node (Person/Group model), so will need to consider how to align this w/ Django users
- "Name" can be remodeled to be a single string rather than labels as with concepts, which will simplify assignment here

**Comment by robgaston** [comment]
@Habennin I would like to make sure that we get the semantics right in the "name" node change mentioned above.  Is this something you could take a look at? please feel free to use our hosted CI application for this.  thanks!

**Comment by Habennin** [comment]
@robgaston I don't see the motivation for changing the name to a simple label. I'm pretty sure that there will be multilinguality etc. Since you will need to have a widget or form or whatever to ahndle this in concept, why would you make another one that doesn't something different? When this data gets published out it will have to follow the form I already put in here, so it seems to add unnecessary complexity to 'simplify' it.

Re scheme owner, I'm not sure that this is necessarily true what you argue. If the user logged in is Phil he is not the scheme owner. English Heritage is. Or he may be working for Heritage Trust or whatever. I don't see how the logged in user relates.

We definitely need a full group and person model with data managed. 

Regarding status, I would think that this might also have to be published out to the world so not having it as explicit semantic data might be a problem. @Philbox ?

**Comment by Philbox** [comment]
@robgaston I was always under the impression that we'd need to be able to record instances of people/organizations/bibiliographic sources and possible others in lingo so we could use resource instance look-ups for the owner/warrant source nodes at least. If Lingo is a self contained package then won't we need to hold the resource models for those in Lingo?

**Comment by robgaston** [comment]
@Philbox yes, we have resource models for these, however in our discussions about MVP it was determined that we would not need to build custom interfaces into Lingo (for MVP at least) as these entities could be managed in the the Arches UI which can also be available to Lingo users.  We can of course build custom interfaces for managing these in Lingo, but it is not currently on the MVP roadmap.

**Comment by Philbox** [comment]
@robgaston If in Lingo I click on the dropdown for Scheme Owner will I get the option to 'Create a new Person', 'Create a new Organization'? If I do then great. I'm not really bothered about whether it lives in Lingo or the Arches UI as long as I can add new instances.
